### DISCUSSION TITLE
Support to set cache-control in Hue response header for hue 4.8

### DISCUSSION THREAD

**Discussion by michelle37** [cloudera/hue#3309] [Questions] [answered]
This issue was an open case https://issues.cloudera.org/browse/HUE-9248 but it was never been resolved, we have a security vulnerability due to the incorrect catch-control resulting concerns for attacker to view sensitive information

Describe the bug:
The Hue application allows developers to submit SQL-like queries against large datasets managed by the
Hadoop/EMR stack. Vanguard plans on using the Hue application to view and transform data that may contain PII or
other sensitive data, however the Hue application does not make use of Cache-Control headers that prevent those
results from being cached by the user's browser.

Steps to reproduce it?
we have tried to login hue 4.4 and ran some queries with PII data, we did see some security policy alert:
There are other HTTP response headers that can be controlled via hue.ini such as secure_content_security_policy, secure_browser_xss_filter, secure_content_type_nosniff etc. However, there is no such setting available for cache-control header.

Hue version or source?
This was occurred in Hue 4.4, can I get a confirmation on version 4.8 or higher? has this been worked on since the very first ticket opened a few years ago?

**Comment by Harshg999** [comment, marked as answer]
Hey @michelle37, thanks for flagging this issue!
Recently @athithyaaselvam has been working on a related issue and the PR is up for review: https://github.com/cloudera/hue/pull/3296

I feel the PR also solves this issue and now we are introducing a config to disable cache and adding headers to restrict caching. (There can be performance issues in Hue if enabled)

It will be available in the master soon and in a subsequent release in the near future.
